activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/converter/EventConverterContext.java

```diff
@@ -70,35 +77,64 @@ public ProcessEngineEvent from(ActivitiEvent activitiEvent) {
     }
 
     public static String getPrefix(ActivitiEvent activitiEvent) {
+        if (isProcessEvent(activitiEvent)) {
+            return PROCESS_EVENT_PREFIX;
+        }
+
+        if (isTaskEvent(activitiEvent)) {
+            return TASK_EVENT_PREFIX;
+        }
+
+        if (isIdentityLinkEntityEvent(activitiEvent)) {
+            IdentityLink identityLinkEntity = (IdentityLink) ((ActivitiEntityEvent) activitiEvent).getEntity();
+            if (isCandidateUserEntity(identityLinkEntity)) {
+                return TASK_CANDIDATE_USER_EVENT_PREFIX;
+            }
+
+            if (isCandidateGroupEntity(identityLinkEntity)) {
+                return TASK_CANDIDATE_GROUP_EVENT_PREFIX;
+            }
+        }
+
+        return ACTIVITI_EVENT_PREFIX;
+    }
+
+    private static boolean isProcessEvent(ActivitiEvent activitiEvent) {
         if (activitiEvent instanceof ActivitiEntityEvent) {
             Object entity = ((ActivitiEntityEvent) activitiEvent).getEntity();
-            if (entity != null) {
-                if (ProcessInstance.class.isAssignableFrom(entity.getClass())) {
-                    if (activitiEvent.getType().equals(ActivitiEventType.ENTITY_SUSPENDED) ||
-                            activitiEvent.getType().equals(ActivitiEventType.ENTITY_ACTIVATED) ||
-                            activitiEvent.getType().equals(ActivitiEventType.ENTITY_CREATED)) {
-                        ExecutionEntity executionEntity = (ExecutionEntity) entity;
-                        if (executionEntity.isProcessInstanceType()) {
-                            return "ProcessInstance:";
-                        } else {
-                            return "";
-                        }
-                    }
-                    return "ProcessInstance:";
-                } else if (entity instanceof Task) {
-                    return "Task:";
-                } else if (entity instanceof IdentityLink){
-                    IdentityLink identityLink = (IdentityLink)entity;
-                    if (IdentityLinkType.CANDIDATE.equalsIgnoreCase(identityLink.getType()) && identityLink.getUserId() != null) {
-                        return "TaskCandidateUser:";
-                    } else if (IdentityLinkType.CANDIDATE.equalsIgnoreCase(identityLink.getType()) && identityLink.getGroupId() != null) {
-                        return "TaskCandidateGroup:";
-                    } else {
-                        return "";
-                    }
-                }
+            if (entity != null && ProcessInstance.class.isAssignableFrom(entity.getClass())) {
+                return isExecutionEntityEvent(activitiEvent) ?
+                        ((ExecutionEntity) entity).isProcessInstanceType() :
+                        true;
             }
         }
-        return "";
+
+        return activitiEvent.getType() == ActivitiEventType.PROCESS_CANCELLED;
```

Comment:
@constantin-ciobotaru this is not clear to me.. why do we need to do this boolean check in there? 
Let's do it in an if block and assign the value to a boolean variable that we return. If tomorrow there is another ActivitiEventType which is process related, we will need to add it there anyways..

Reply:
1. It seems that the event fired when deleting a process instance is ActivitiProcessCancelledEventImpl which is an ActivitiEvent not an ActivitiEntityEvent, so it should be treated separately.
2. Good idea with the 'if', it will be more clear.